src/python/bot/tasks/progression_task.py

```diff
@@ -256,6 +256,10 @@ def find_fixed_range(testcase_id, job_type):
   if not max_revision:
     max_revision = revisions.get_last_revision_in_list(revision_list)
 
+  # If the min and max revisions are the same, there's no work to do.
+  if min_revision == max_revision:
```

Comment:
I don't think it was happening here.
Current regression range was like 284921838:285129628. But then it ran into the loop, and somehow min==max. i am suspecting this
      # Skip this revision.
      del revision_list[middle_index]
      max_index -= 1
      continue ?

Reply:
Discussed offline but I don't think that's the cause either. I just ended up explicitly handling the case where the revisions are equal.